front/components/data_source/DataSourceEdition.tsx

```diff
@@ -0,0 +1,324 @@
+import {
+  Avatar,
+  Button,
+  CloudArrowLeftRightIcon,
+  Dialog,
+  Icon,
+  LockIcon,
+  Modal,
+  Page,
+} from "@dust-tt/sparkle";
+import type {
+  ConnectorProvider,
+  ConnectorType,
+  EditedByUser,
+  UpdateConnectorRequestBody,
+  UserType,
+  WhitelistableFeature,
+  WorkspaceType,
+} from "@dust-tt/types";
+import { CONNECTOR_TYPE_TO_MISMATCH_ERROR } from "@dust-tt/types";
+import { InformationCircleIcon } from "@heroicons/react/20/solid";
+import type { NextRouter } from "next/router";
+import React, { useContext, useState } from "react";
+
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import { CONNECTOR_CONFIGURATIONS } from "@app/lib/connector_providers";
+import { formatTimestampToFriendlyDate } from "@app/lib/utils";
+import { setupConnection } from "@app/pages/w/[wId]/builder/data-sources/managed";
+
+export type DataSourceIntegration = {
+  name: string;
+  dataSourceName: string | null;
+  connector: ConnectorType | null;
+  fetchConnectorError: boolean;
+  fetchConnectorErrorMessage?: string | null;
+  status: "preview" | "built" | "rolling_out";
+  rollingOutFlag: WhitelistableFeature | null;
+  connectorProvider: ConnectorProvider;
+  description: string;
+  limitations: string | null;
+  guideLink: string | null;
+  synchronizedAgo: string | null;
+  setupWithSuffix: string | null;
+  usage: number | null;
+  editedByUser?: EditedByUser | null;
+};
+
+interface DataSourceEditionModalProps {
+  isOpen: boolean;
+  owner: WorkspaceType;
+  connectorProvider: ConnectorProvider | null | undefined;
+  dataSourceIntegration: DataSourceIntegration | null;
+  onClose: () => void;
+  router: NextRouter;
+  dustClientFacingUrl: string;
+  user: UserType;
+  setIsRequestDataSourceModalOpen: (show: boolean) => void;
+}
+
+export function DataSourceEditionModal({
```

Comment:
What do we do with `DataSourceDetailsModal` which is the current implementation of this (in a simpler version)?

Reply:
Looking at the Figma, is seems to still exist.. IMHO the only way to edit the permissions should be through this `DataSourceEditionModal` and not when selecting data